inotify/adapters.py

```diff
@@ -227,53 +227,55 @@ def event_gen(self):
 
 class InotifyTree(BaseTree):
     def __init__(self, path, mask=inotify.constants.IN_ALL_EVENTS,
-                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S):
+                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S, ignored_dirs=[]):
         super(InotifyTree, self).__init__(mask=mask, block_duration_s=block_duration_s)
```

Comment:
You should push ignored_dirs down to (_)BaseTree's __init__ and store it on object from there as it needs to be referenced there (line 216).